pkg/descheduler/framework/plugins/loadaware/utilization_util.go

```diff
@@ -243,40 +312,115 @@ func evictPodsFromSourceNodes(
+	bothTotalAvailableUsage, bothTotalNodes := targetAvailableUsage(bothDestinationNodes, resourceNames, false)
+	prodBothAvailableUsage, prodBothTotalNodes := targetAvailableUsage(bothDestinationNodes, resourceNames, true)
+	klog.V(4).InfoS("node pool availableUsage", "onlyNodeTotal", totalAvailableUsages, "onlyProdOnly", prodAvailableUsages,
+		"bothLowNodesTotal", bothTotalAvailableUsage, "bothLowProdTotal", prodBothAvailableUsage)
+
+	nodeTotalAvailableUsages := newAvailableUsage(resourceNames)
+	for _, resourceName := range resourceNames {
+		if quantity, ok := nodeTotalAvailableUsages[resourceName]; ok {
+			if _, totalOk := totalAvailableUsages[resourceName]; totalOk {
+				quantity.Add(*totalAvailableUsages[resourceName])
+			}
+			if _, prodOk := bothTotalAvailableUsage[resourceName]; prodOk {
+				quantity.Add(*bothTotalAvailableUsage[resourceName])
+			}
+		}
+	}
+	nodeKeysAndValues := []interface{}{
+		"nodePool", nodePoolName,
+	}
+	for resourceName, quantity := range nodeTotalAvailableUsages {
+		nodeKeysAndValues = append(nodeKeysAndValues, string(resourceName), quantity.String())
+	}
+	klog.V(4).InfoS("Total node usage capacity to be moved", nodeKeysAndValues...)
 
-		for _, resourceName := range resourceNames {
-			quantity, ok := totalAvailableUsages[resourceName]
-			if !ok {
-				switch resourceName {
-				case corev1.ResourceCPU:
-					quantity = resource.NewMilliQuantity(0, resource.DecimalSI)
-				case corev1.ResourceMemory, corev1.ResourceEphemeralStorage, corev1.ResourceStorage:
-					quantity = resource.NewQuantity(0, resource.BinarySI)
-				default:
-					quantity = resource.NewQuantity(0, resource.DecimalSI)
-				}
-				totalAvailableUsages[resourceName] = quantity
+	targetNodes = append(targetNodes, bothTotalNodes...)
+	balancePods(ctx, nodePoolName, sourceNodes, targetNodes, nodeTotalAvailableUsages, dryRun, nodeFit, false, resourceWeights, podEvictor,
+		podFilter, nodeIndexer, continueEviction, evictionReasonGenerator)
+
+	// bothLowNode will be used by nodeHigh and prodHigh nodes, needs sub resources used by pods on nodeHigh.
+	for _, resourceName := range resourceNames {
+		if quantity, ok := nodeTotalAvailableUsages[resourceName]; ok {
+			if bothTotalAvailableUsage[resourceName].Cmp(*quantity) > 1 {
+				bothTotalAvailableUsage[resourceName].Sub(*quantity)
+			} else {
+				// all resource used by nodeHigh pods
+				bothTotalAvailableUsage[resourceName].Sub(*bothTotalAvailableUsage[resourceName])
 			}
-			quantity.Add(*destinationNode.thresholds.highResourceThreshold[resourceName])
-			quantity.Sub(*destinationNode.usage[resourceName])
 		}
 	}
 
-	keysAndValues := []interface{}{
+	prodTotalAvailableUsages := newAvailableUsage(resourceNames)
+	for _, resourceName := range resourceNames {
+		if quantity, ok := prodTotalAvailableUsages[resourceName]; ok {
+			if _, totalOk := prodAvailableUsages[resourceName]; totalOk {
+				quantity.Add(*prodAvailableUsages[resourceName])
```

Comment:
These variables' name is not clear, consider rename them? For example, totalOk is used to judge prodAvailableUsage?